presto-parser/src/main/java/com/facebook/presto/sql/tree/QualifiedName.java

```diff
@@ -32,6 +32,9 @@
     private final List<String> parts;
     private final List<String> originalParts;
 
+    private Optional<QualifiedName> cachedPrefix;
+    private String cachedSuffix;
```

Comment:
@martint  I could set these two variables as `final` and initialize them in its constructor.